### PR TITLE
+added task_22

### PR DESCRIPTION
**Pull request by SlavaPeshko**
Task_22

### CODE REVIEW

File: 1-select-tasks/22.txt [hunk 1]
```diff
@@ -1,3 +1,15 @@
...
-| CompanyName | ProductName | PricePerItem |
\ No newline at end of file
+--| CompanyName | ProductName | PricePerItem |
+
+SELECT [c].[CompanyName], [p].[ProductName], MAX([p].[UnitPrice])  AS [PricePerItem]
+FROM [dbo].[Customers][c]
+JOIN [dbo].[Orders][o] ON [o].[CustomerID] = [c].[CustomerID]
+JOIN [dbo].[Order Details][od] ON [od].[OrderID] = [o].[OrderID]
+JOIN [dbo].[Products][p] ON [p].[ProductID] = [od].[ProductID]
+GROUP BY [c].[CustomerID], [c].[CompanyName],[p].[ProductName], [p].[UnitPrice]
```
Comment (AlexMokin) [inline 1-select-tasks/22.txt]: what do you need the group by clause for?

Comment (SlavaPeshko) [reply 1-select-tasks/22.txt]: before HAVING use, and used aggregate functions.

**Comment by SlavaPeshko** [comment]
Update task_22

File: 1-select-tasks/22.txt [hunk 2]
```diff
@@ -1,3 +1,14 @@
...
+--| CompanyName | ProductName | PricePerItem |
+
+SELECT [c].[CustomerID], [c].[CompanyName], [p].[ProductName], [od].[UnitPrice]  AS [PricePerItem]
+FROM [dbo].[Customers][c]
+JOIN [dbo].[Orders][o] ON [o].[CustomerID] = [c].[CustomerID]
+JOIN [dbo].[Order Details][od] ON [od].[OrderID] = [o].[OrderID]
+JOIN [dbo].[Products][p] ON [p].[ProductID] = [od].[ProductID]
+WHERE [od].[UnitPrice] = (SELECT MAX([od2].[UnitPrice]) FROM [dbo].[Order Details][od2]
+		JOIN [dbo].[Orders] [o2] ON [o2].[OrderID] = [od2].[OrderID]
+		JOIN [dbo].[Products][p2] ON [p2].[ProductID] = [od2].[ProductID]
```
Comment (AlexMokin) [inline 1-select-tasks/22.txt]: you don't use p2 table

**Comment by SlavaPeshko** [comment]
*update task_22

**Comment by AlexMokin** [comment]
done

File: 1-select-tasks/22.txt [hunk 3]
```diff
@@ -1,3 +1,14 @@
-Create a SQL query to display the name of each customer along with their most expensive purchased product:
+--22Create a SQL query to display the name of each customer along with their most expensive purchased product:
 
-| CompanyName | ProductName | PricePerItem |
\ No newline at end of file
+--| CompanyName | ProductName | PricePerItem |
+
+SELECT [c].[CustomerID], [c].[CompanyName], [p].[ProductName], [od].[UnitPrice]  AS [PricePerItem]
```
Comment (AlexMokin) [inline 1-select-tasks/22.txt]: There are duplicates in the result

**Pull request closed without merging**